networking/network_policy/about-network-policy.adoc

```diff
@@ -5,11 +5,584 @@ include::modules/common-attributes.adoc[]
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: deny-all-ingress
+spec:
+  podSelector: {}
+  ingress: []
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-all-https
+spec:
+  podSelector: {}
+  ingress:
+  - ports:
+    - port: 443
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-only-db-https
+spec:
+  podSelector: {}
+  ingress:
+    from:
+    - namespaceSelector:
+        matchLabels:
+          app: db
+      ports:
+      - port: 443
+----
+////
+
+[id="policy-rule-behavior-with-an-empty-ingress-or-egress-rule-set_{context}"]
+=== Behavior with an empty ingress or egress rule set
+
+If you define a network policy rule with an empty list, the network policy does allow any traffic. Any pods selected by the network policy are completely isolated.
+
+For example, the following ingress rule set selects no ingress traffic:
+
+.Example ingress ruleset defined as an empty list
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: <name>
+spec:
+  ingress: []
+----
+
+////
+Name:         name1
+Namespace:    default
+Created on:   2021-06-28 21:11:24 -0400 EDT
+Labels:       <none>
+Annotations:  <none>
+Spec:
+  PodSelector:     <none> (Allowing the specific traffic to all pods in this namespace)
+  Allowing ingress traffic:
+    <none> (Selected pods are isolated for ingress connectivity)
+  Not affecting egress traffic
+  Policy Types: Ingress
+////
+
+An empty rule set is the inverse of a rule set with a single empty selector, which selects all traffic. For example, the following ingress rule set selects all ingress traffic:
+
+.Example ingress ruleset with a single empty selector
+[source,yaml]
+----
+kind: NetworkPolicy # Valid
+apiVersion: networking.k8s.io/v1
+spec:
+  ingress:
+  - {}
+----
+
+////
+Spec:
+  PodSelector:     <none> (Allowing the specific traffic to all pods in this namespace)
+  Allowing ingress traffic:
+    To Port: <any> (traffic allowed to all ports)
+    From: <any> (traffic not restricted by source)
+  Not affecting egress traffic
+  Policy Types: Ingress
+////
+
+The previous example is identical in behavior to the following example:
+
+.Example ingress ruleset that matches all ingress traffic
+[source,yaml]
+----
+kind: NetworkPolicy # Valid
+apiVersion: networking.k8s.io/v1
+spec:
+  ingress:
+  - from:
+    - podSelector: {}
+      namespaceSelector: {}
+----
+
+////
+Spec:
+  PodSelector:     <none> (Allowing the specific traffic to all pods in this namespace)
+  Allowing ingress traffic:
+    To Port: <any> (traffic allowed to all ports)
+    From:
+      NamespaceSelector: <none>
+      PodSelector: <none>
+  Not affecting egress traffic
+  Policy Types: Ingress
+////
+
+[id="nework-policy-api_{context}"]
+== Network policy API
+
+When you specify network policy, you use the `NetworkPolicy` API in the `networking.k8s.io` API group.
+
+TODO: Various network policy API tables.
```

Comment:
network policy api tables?

Reply:
@astoycos, oh, yeah, so this is probably going to be all the valid fields for the NP API.